### PR TITLE
[WIP] - updating project and example to rn 0.60.3

### PR DESCRIPTION
**Pull request by MateusAndrade**
Since react-native 0.60 has landed, some library's need to be updated to support auto-linking, and other features(hermes, etc).

This PRS update the example project to 0.60.3, and update the peerDependencies to >= 0.60.0 followed by a major version to 4.0.0.

Currently, im running some test on Android emulator using the example without any problem. However, i still need to test some changes on IOS, using this project on a project using react-native <= 0.59, etc.

## Pending steps:

- [ ] Test android version on release mode
- [ ] Update IOS version
- [ ] Test IOS version on archive/dev mode
- [ ] Test this update on a project using react-native <= 0.59 with jetifier

### CODE REVIEW

File: examples/android/app/build.gradle [hunk 1]
```diff
@@ -134,13 +139,32 @@ android {
...
+
+
 }
 
 dependencies {
-    compile project(':react-native-splash-screen')
-    compile fileTree(dir: "libs", include: ["*.jar"])
-    compile "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
-    compile "com.facebook.react:react-native:+"  // From node_modules
+    implementation project(':react-native-splash-screen')
```
Comment (sibelius) [inline examples/android/app/build.gradle]: can't autolinking handle this?

File: examples/android/app/src/main/java/com/examples/MainApplication.java [hunk 2]
```diff
@@ -21,10 +23,10 @@ public boolean getUseDeveloperSupport() {
     }
 
     @Override
-    protected List<ReactPackage> getPackages() {
+      protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
           new MainReactPackage(),
-            new SplashScreenReactPackage()
+          new SplashScreenReactPackage()
```
Comment (sibelius) [inline examples/android/app/src/main/java/com/examples/MainApplication.java]: autolinking?

File: examples/android/settings.gradle [hunk 3]
```diff
@@ -1,5 +1,7 @@
 rootProject.name = 'examples'
+apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
+
 include ':react-native-splash-screen'
-project(':react-native-splash-screen').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-splash-screen/android')
+project(':react-native-splash-screen').projectDir = new File(rootProject.projectDir, '../../android')
```
Comment (sibelius) [inline examples/android/settings.gradle]: autolinking?

File: examples/metro.config.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    getTransformOptions: async () => ({
+      transform: {
+        experimentalImportSupport: false,
+        inlineRequires: false
+      }
+    })
+  },
+  watchFolders: [
+    path.resolve(__dirname, "node_modules"),
+    reactNativeSplachScreenRoot
```
Comment (sibelius) [inline examples/metro.config.js]: nice

**Comment by radko93** [comment]
@MateusAndrade hey, are you going to work further on this? Just asking as I'm gonna need this so not sure if to start myself

File: examples/metro.config.js (hunk 4 above)
Comment (mAiNiNfEcTiOn) [reply examples/metro.config.js]: Typo: where it is _Splach_ should be _Splash_

**Comment by MateusAndrade** [comment]
> this

You can handle this? Unfortunately i got really busy on the last months and can give a proper look into this. Please, let me know if can help you with anything.

**Pull request closed without merging**